### PR TITLE
update to waf 2.0.19

### PR DESCRIPTION
**Pull request by cosnicolaou**
This PR is a step to getting all of the tests working again. The only outstanding problem is that
boost tests don't work on a mac, but they do on linux. The problem is that waf configure scripts test for the presence of boost using the normal python flags, but the actual build of the tests uses the python-config --include and --ldflags values which do not have the correct include path for boost. On linux the two sets of flags are identical.

### CODE REVIEW

**Comment by gjcarneiro** [comment]
looks promising; let me know when it's ready to review/merge

**Comment by cosnicolaou** [comment]
You can go ahead and review it now if you wish. Do you have any suggestions
for how to handle the boost on Mac problem?

Cheers, Cos.

**Comment by gjcarneiro** [comment]
I don't have a Mac.  Maybe this?

```
diff --git a/wscript b/wscript
index a2a96b1..a68f417 100644
--- a/wscript
+++ b/wscript
@@ -187,7 +187,14 @@ def configure(conf):
         if not conf.check_nonfatal(header_name='stdint.h'):
             conf.env.append_value('CPPPATH', os.path.join(conf.curdir, 'include'))
 
-    if conf.check_nonfatal(header_name='boost/shared_ptr.hpp'):
+    conf.env.stash()
+    try:
+        conf.env.append_value("CXXFLAGS", conf.env["CXXFLAGS_PYEXT"])
+        conf.env.append_value("INCLUDES", conf.env["INCLUDES_PYEXT"])
+        have_boost = conf.check_nonfatal(header_name='boost/shared_ptr.hpp')
+    finally:
+        conf.env.revert()
+    if have_boost:
         conf.env['ENABLE_BOOST_SHARED_PTR'] = True
 
     conf.recurse('benchmarks')
```

File: waf-tools/python_patched.py [hunk 1]
```diff
@@ -241,7 +242,12 @@ def check_python_headers(conf):
 	if not pybin:
 		conf.fatal('Could not find the python executable')
 
-	v = 'prefix SO LDFLAGS LIBDIR LIBPL INCLUDEPY Py_ENABLE_SHARED MACOSX_DEPLOYMENT_TARGET LDSHARED CFLAGS LDVERSION'.split()
```
Comment (gjcarneiro) [inline waf-tools/python_patched.py]: I don't remember anymore why this `python_patched` module exists.  This is based on waf's own `python` "tool".  I might have applied some patch to it, to fix what, I can't remember anymore, been so long.

Perhaps it is worth checking that the normal `python` tool in WAF works as expected without any patching?

**Comment by gjcarneiro** [comment]
OK, I had a minor comment, but nothing major, so, might as well merge it as it is now, and if you want to improve on it, you can submit another PR.

**Pull request merged**